contracts/modules/assessment/Assessment.sol

```diff
@@ -2,515 +2,365 @@
-  ///                       those that are true represent an accept vote.
-  /// @param stakeIncrease  When a non-zero value is given, this function will also increase the
-  ///                       stake in the same transaction.
-  function castVotes(
-    uint[] calldata assessmentIds,
-    bool[] calldata votes,
-    string[] calldata ipfsAssessmentDataHashes,
-    uint96 stakeIncrease
-  ) external override onlyMember whenNotPaused {
-    require(assessmentIds.length == votes.length, AssessmentIdsVotesLengthMismatch());
-    require(assessmentIds.length == ipfsAssessmentDataHashes.length, AssessmentIdsIpfsLengthMismatch());
-
-    if (stakeIncrease > 0) {
-      stake(stakeIncrease);
-    }
+  /// @notice Returns the payout cooldown period for a given product type
+  /// @param productTypeId The product type identifier
+  /// @return The cooldown period in seconds
+  function payoutCooldown(uint productTypeId) override external view returns (uint) {
+    AssessmentData memory assessmentData = _assessmentData[productTypeId];
+    require(assessmentData.assessingGroupId != 0, InvalidProductType());
 
-    for (uint i = 0; i < assessmentIds.length; i++) {
-      _castVote(assessmentIds[i], votes[i], ipfsAssessmentDataHashes[i]);
-    }
+    return assessmentData.cooldownPeriod;
   }
 
-  /// Casts a vote on an assessment
-  ///
-  /// @dev Resets the poll's end date on the first vote. The first vote can only be an accept vote.
-  /// If no votes are cast during minVotingPeriodInDays it is automatically considered denied. When
-  /// the poll ends in less than silentEndingPeriodInDays, the end date is extended with a potion of
-  /// silentEndingPeriodInDays proportional to the user's stake compared to the
-  /// total stake on that assessment, namely the sum of tokens used for both accept and deny votes,
-  /// but no greater than silentEndingPeriodInDays.
-  ///
-  /// @param assessmentId  The index of the assessment for which the vote is cast
-  /// @param isAcceptVote  True to accept, false to deny
-  function _castVote(uint assessmentId, bool isAcceptVote, string memory ipfsAssessmentDataHash) internal {
-
-    {
-      require(!hasAlreadyVotedOn[msg.sender][assessmentId], AlreadyVoted());
-      hasAlreadyVotedOn[msg.sender][assessmentId] = true;
-    }
+  /// @notice Returns the assessor group ID for a given claim
+  /// @param claimId The claim identifier
+  /// @return The group ID of the assessors for the claim
+  function assessorGroupOf(uint claimId) override external view returns (uint32) {
+    Assessment memory assessment = _assessments[claimId];
+    require(assessment.start != 0, InvalidClaimId());
 
-    uint96 stakeAmount = stakeOf[msg.sender].amount;
-    require(stakeAmount > 0, StakeRequired());
+    return assessment.assessingGroupId;
+  }
+
+  function getAssessment(uint claimId) override external view returns(Assessment memory assessment) {
+    return _assessments[claimId];
```

Comment:
should it revert if the assessment doesn't exist?

Reply:
actually I'm tempted to make the viewer functions to not revert and just return default values if it does not exist.

this might be friendlier for the FE so that they can do multicall on the view functions safely and just interpret the results

so in `assessorGroupOf` i'll drop the require:
<img width="597" alt="Screenshot 2025-07-10 at 16 28 20" src="https://github.com/user-attachments/assets/6b67b0b7-b2c2-40f1-abe9-6ff2adb2ec96" />